openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -344,29 +345,74 @@ def post(self, request):
         Marks the user as having no password set for deactivation purposes,
         and logs the user out.
         """
-        username = None
         user_model = get_user_model()
         try:
             # Get the username from the request and check that it exists
-            username = request.data['username']
-            user = user_model.objects.get(username=username)
-
-            with transaction.atomic():
-                # 1. Unlink LMS social auth accounts
-                UserSocialAuth.objects.filter(user_id=user.id).delete()
-                # 2. Change LMS password & email
-                user.email = get_retired_email_by_email(user.email)
-                user.save()
-                _set_unusable_password(user)
-                # 3. Unlink social accounts & change password on each IDA, still to be implemented
-            return Response(status=status.HTTP_204_NO_CONTENT)
+            verify_user_password_response = self._verify_user_password(request)
+            if verify_user_password_response.status_code == status.HTTP_204_NO_CONTENT:
```

Comment:
Instead of checking `==`, why not do
```
if verify_user_password_response.status_code != status.HTTP_204_NO_CONTENT:
    return verify_user_password_response
...
```

Reply:
It doesn't look like `_verify_user_password` will ever throw? Should it be executed before the `try/except`?